compiler/rustc_ast/Cargo.toml

```diff
@@ -7,7 +7,7 @@ edition = "2021"
 # FIXME: bumping memchr to 2.7.1 causes linker errors in MSVC thin-lto
 # tidy-alphabetical-start
 bitflags = "2.4.1"
-memchr = "=2.5.0"
```

Comment:
This is necessary, please see the above comment. https://github.com/rust-lang/rust/pull/121465